mono/utils/mono-threads.h

```diff
@@ -153,6 +153,8 @@ enum {
 	ASYNC_SUSPEND_STATE_INDEX = 1,
 };
 
+struct HandleStack;
```

Comment:
Those are types in metadata and sgen, they should not be in utils at all.

Reply:
Forward declaring a struct doesn’t really expose anything. It is still opaque. We expose the presence of the pointers in the thread already.

Reply:
This is not the style we have in mono and this is violating basic layering.

Reply:
The layering is preserved as the types are opaque.
The layering is already similarly violated/preserved by putting the pointers in utils, albeit void*. Right? Why is this worse than that?
It helps debugging and cleans up the C code. Seems like a win.